src/protocols/rdp/rdp.c

```diff
@@ -537,6 +536,7 @@ static int guac_rdp_handle_connection(guac_client* client) {
     }
 
     /* Upgrade to write lock again for further exclusive operations */
+    guac_rwlock_release_lock(&(rdp_client->lock));
```

Comment:
This shouldn't be necessary - the guac rwlock is set up to handle upgrading from a read lock to a write lock just by calling `guac_rwlock_acquire_write_lock`. I don't think coverity fully understands how all this works (either that, or _I'm_ confused).

Reply:
From what I read it seems like upgrading to a write lock adds a layer on top of the read locks and the still requires a free for each layer. I figured it was more straight forward an not necessary in this case to layer the locks

Reply:
Nevermind, I am confused.